developer/guides/components/demo-component.mdx

```diff
@@ -0,0 +1,559 @@
+  ></video>
+</Frame>
+
+To make this a more realistic example, we can create an array of accordion items, then use `accordions.map` to create an `<Accordion.Item>` for each one:
+
+```tsx src/components/Accordions/index.tsx {5-14, 18-21}
+import * as Accordion from "@radix-ui/react-accordion";
+
+export function Accordions() {
+  const accordions = [
+    {
+      title: "Accordion 1",
+      content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
+    },
+    {
+      title: "Accordion 2",
+      content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
+    },
+  ];
+
+  return (
+    <Accordion.Root type="single">
+      {accordions.map((accordion, i) => (
+        <Accordion.Item key={i} value={`item-${i}`}>
+          <Accordion.Trigger>{accordion.title}</Accordion.Trigger>
+          <Accordion.Content>{accordion.content}</Accordion.Content>
+        </Accordion.Item>
+      ))}
+    </Accordion.Root>
+  );
+}
+```
+
+## Make the component editable in Makeswift
+
+Our current example hardcodes the accordion content. We’re now going to make our component editable in Makeswift by using controls to accept user input for each accordion item.
+
+### Accordion items
+
+We can access the Makeswift controls and have them return input from the user in Makeswift when we register the component in the `Accordions.makeswift.ts` file. Each control adds properties to the component that we can edit in Makeswift.
+
+We want the user to be able to edit the title for an accordion item. For this, we can use the [`TextInput`](/developer/reference/controls/text-input) control. We'll import this into the `Accordions.makeswift.ts` file and add it to our `props` object.
+
+```tsx src/components/Accordions/Accordions.makeswift.tsx {1, 11-14}
+import { TextInput } from "@makeswift/runtime/controls";
+
+import { runtime } from "src/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {
+    accordions: TextInput({
+      label: "Title",
+      defaultValue: "Accordion title",
+    }),
+  },
+});
+```
+
+We also want the user to be able to add multiple accordion items, not just one. To support this, we can use the [`List`](/developer/reference/controls/list) control and put the `title` prop inside of it.
+
+```tsx src/components/Accordions/Accordions.makeswift.tsx {1, 11-17}
+import { TextInput, List } from "@makeswift/runtime/controls";
+
+import { runtime } from "@/makeswift/runtime";
+
+import { Accordions } from "./";
+
+runtime.registerComponent(Accordions, {
+  type: "accordions",
+  label: "Accordions",
+  props: {
+    accordions: List({
+      label: "Accordions",
+      type: TextInput({ label: "Title", defaultValue: "Accordion Title" }),
+      getItemLabel(accordionItem) {
+        return accordionItem ?? "Accordion Title";
+      },
+    }),
+  },
+});
+```
+
```

Comment:
This is already a lot to process. I'd summary / have a video demo of what we've done so far here.